libsolidity/formal/Predicate.cpp

```diff
@@ -49,26 +49,25 @@ Predicate const* Predicate::create(
 	std::vector<ScopeOpener const*> _scopeStack
 )
 {
-	smt::SymbolicFunctionVariable predicate{_sort, std::move(_name), _context};
-	std::string functorName = predicate.currentName();
-	solAssert(!m_predicates.count(functorName), "");
+	solAssert(!m_predicates.count(_name), "");
 	return &m_predicates.emplace(
 		std::piecewise_construct,
```

Comment:
Ok, here we assert that there's only one predicate per name, then the ``.emplace`` is in fact the only place, in which Predicate's are constructed - and ``newFunctor`` was previously fully unused?
If all that is true, then this seems sound.
I vaguely remember that there was some reason for this in the distant past, but maybe I'm misremembering and/or things changed since then :-).

Reply:
Yes, yes and yes :)
The one predicate per name assert was also there before this change.
`newFunctor` was indeed not used anywhere.